app/main.py

```diff
@@ -1,8 +1,28 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_instances = []
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data["age"]
+        person = Person(name, age)
+        person_instances.append(person)
+    for person_data in people:
+        name = person_data["name"]
+        person_instance = Person.people[name]
+
+        spouse = person_data.get("wife") or person_data.get("husband")
+        if spouse:
+            if "wife" in person_data:
+                person_instance.wife = Person.people[spouse]
+            elif "husband" in person_data:
+                person_instance.husband = Person.people[spouse]
```

Comment:
The code for setting the `wife` and `husband` properties is a bit confusing. You are checking if `wife` or `husband` is in the `person_data` dictionary, and then you are checking again if `wife` or `husband` is in the dictionary. Consider simplifying this code. You could, for example, use a single `if` statement with the `in` operator to check if `wife` or `husband` is in the dictionary, and then set the corresponding property.